packages/amplify-category-geo/src/service-stacks/deviceLocationTrackingStack.ts

```diff
@@ -0,0 +1,173 @@
+        effect: Effect.ALLOW,
+      });
+      geoCreateTrackingStatement.addActions('geo:CreateTracker');
+      geoCreateTrackingStatement.addAllResources();
+
+      const trackingARN = cdk.Fn.sub('arn:aws:geo:${region}:${account}:tracker/${trackerName}', {
+        region: this.trackingRegion,
+        account: cdk.Fn.ref('AWS::AccountId'),
+        trackerName: this.trackerName,
+      });
+
+      const geoUpdateDeleteTrackingStatement = new iam.PolicyStatement({
+        effect: Effect.ALLOW,
+      });
+      geoUpdateDeleteTrackingStatement.addActions('geo:UpdateTracker', 'geo:DeleteTracker');
+      geoUpdateDeleteTrackingStatement.addResources(trackingARN);
+
+      // set up custom params
+      // ....
+      // ....
+
+      const customTrackingLambdaCode = fs.readFileSync(customDeviceLocationTrackingLambdaCodePath, 'utf-8');
+      const customTrackingLambda = new lambda.Function(this, 'customTrackingLambda', {
+        code: lambda.Code.fromInline(customTrackingLambdaCode),
+        handler: 'index.handler',
+        runtime: Runtime.NODEJS_14_X,
+        timeout: Duration.seconds(300),
+      });
+      customTrackingLambda.addToRolePolicy(geoCreateTrackingStatement);
+      customTrackingLambda.addToRolePolicy(geoUpdateDeleteTrackingStatement);
+
+      const trackingCustomResource = new cdk.CustomResource(this, 'CustomTracking', {
+        serviceToken: customTrackingLambda.functionArn,
+        resourceType: 'Custom::LambdaCallout',
+        properties: {
+          trackerName: this.trackerName,
+          region: this.trackingRegion,
+          env: cdk.Fn.ref('env'),
+        },
+      });
+
+      return trackingCustomResource;
+    }
+
+    // Grant read-only access to the Tracking Index for Authorized and/or Guest users
+    private constructTrackingPolicyResources(trackingResource: cdk.CustomResource): void {
+      Object.keys(this.roleAndGroupPermissionsMap).forEach((group: string) => {
+        const crudActions: string[] = _.uniq(_.flatten(this.roleAndGroupPermissionsMap[group]
+          .map((permission: string) => deviceLocationTrackingCrudPermissionsMap[permission])));
+
+        // This is a work-around until the TrackingArn is included in the `UpdateTracker` output
+        const outputTrackingArn = cdk.Fn.sub('arn:aws:geo:${region}:${account}:tracker/${trackerName}', {
+          region: this.trackingRegion,
+          account: cdk.Fn.ref('AWS::AccountId'),
+          collectionName: trackingResource.getAtt('TrackerName').toString(),
+        });
+        const policyDocument = new iam.PolicyDocument({
+          statements: [
+            new iam.PolicyStatement({
+              effect: iam.Effect.ALLOW,
+              actions: crudActions,
+              resources: [outputTrackingArn],
+              conditions: {
+                StringLike: {
+                  'geo:DeviceIds': [
+                    '${cognito-identity.amazonaws.com:sub}',
```

Comment:
Just want to verify with the team that this is the correct way of setting up the cognito identity sub